Datasets/DatasetUtilities.swift

```diff
@@ -0,0 +1,112 @@
+// Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+#if canImport(FoundationNetworking)
+    import FoundationNetworking
+#endif
+
+public struct DatasetUtilities {
+    public static let curentWorkingDirectoryURL = URL(
+        fileURLWithPath: FileManager.default.currentDirectoryPath)
+
+    public static func fetchResource(
+        filename: String,
+        remoteRoot: URL,
+        localStorageDirectory: URL = curentWorkingDirectoryURL
+    ) -> Data {
+        print("Loading resource: \(filename)")
+
+        let resource = ResourceDefinition(
+            filename: filename,
+            remoteRoot: remoteRoot,
+            localStorageDirectory: localStorageDirectory)
+
+        let localURL = resource.localURL
+
+        if !FileManager.default.fileExists(atPath: localURL.path) {
+            print(
+                "File does not exist locally at expected path: \(localURL.path) and must be fetched"
+            )
+            fetchFromRemoteAndSave(resource: resource)
+        }
+
+        do {
+            print("Loading local data at: \(localURL.path)")
+            let data = try Data(contentsOf: localURL)
+            print("Succesfully loaded resource: \(filename)")
+            return data
+        } catch {
+            fatalError("Failed to contents of resource: \(localURL)")
+        }
+    }
+
+    struct ResourceDefinition {
+        let filename: String
+        let remoteRoot: URL
+        let localStorageDirectory: URL
+
+        var localURL: URL {
+            localStorageDirectory.appendingPathComponent(filename)
+        }
+
+        var remoteURL: URL {
+            remoteRoot.appendingPathComponent(filename).appendingPathExtension("gz")
+        }
+
+        var archiveURL: URL {
+            localURL.appendingPathExtension("gz")
+        }
+    }
+
+    static func fetchFromRemoteAndSave(resource: ResourceDefinition) {
```

Comment:
The `resource:` argument label is seemingly only repeating the type information `ResourceDefinition`, and it doesn't affect the grammaticality of the phrase at call sites. I'd suggest dropping the argument label.
```suggestion
    static func fetchFromRemoteAndSave(_ resource: ResourceDefinition) {
```